tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2050,12 +2055,16 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
         contTree(finishTransformExpr(targs.head) match {
           case ClassOf(ClassRef(cls)) =>
             cls match {
-              case "B" | "S" | "C" | "I" | "D" => IntLiteral(0)
-              case "L"                         => LongLiteral(0L)
-              case "F"                         => FloatLiteral(0.0f)
-              case "Z"                         => BooleanLiteral(false)
-              case "V"                         => Undefined()
-              case _                           => Null()
+              case "C" => IntLiteral(0) // yes, as an Int
```

Comment:
Because that's what [the user-space implementation](https://github.com/scala-js/scala-js/blob/4093c9bf41607320eae260f9d7dbd3c8cf02fc0d/scalalib/overrides-2.12/scala/collection/mutable/ArrayBuilder.scala#L79) does as well ;)
Ultimately that's because an `Array[Char]` stores primitive JS numbers, so we don't want the JS array being built to contain boxed chars.

I can add a comment.